### PR TITLE
feat: Due to new version of django-cor-headers added two settings in …

### PR DESCRIPTION
**Pull request by awais786**
First Phase
refactor: Due to new version of django-cor-headers added two settings in internal repo and now picking as per installed version.

Internal Repo PR
https://github.com/edx/edx-internal/pull/3434/files

---------------

Second Phase
After merging and deploy above PRs merge this 
https://github.com/edx/course-discovery/pull/2783

---------------
After all above steps we can renamed the settings name
 `DISCOVERY_CORS_ORIGIN_WHITELIST_NEW` to `DISCOVERY_CORS_ORIGIN_WHITELIST` and maintained only 1 list

### CODE REVIEW

**Comment by awais786** [comment]
@jmbowman I have created a PR as per your [suggestion](https://github.com/edx/edx-internal/pull/3434#issuecomment-713021669). Please check.

We have to merge internal config PR and this PR first and deploy on prod. After that we will upgrade the [django-cor-headers.](https://github.com/edx/course-discovery/pull/2783)

**Comment by awais786** [comment]
I have created sandbox with this PR and https://github.com/edx/sandbox-internal/pull/181
Pip freeze shows `django-cors-headers==2.5.3`
and django shell loads the expected settings
```discovery@awais786:~/discovery$ python manage.py shell
2021-08-02 14:22:10,051 INFO 207345 [algoliasearch_django.registration] /edx/app/discovery/venvs/discovery/lib/python3.8/site-packages/algoliasearch_django/registration.py:77 - REGISTER <class 'course_discovery.apps.course_metadata.algolia_models.AlgoliaProxyProduct'>
Python 3.8.10 (default, Jun  2 2021, 10:49:15)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from django.conf import settings
>>> settings.CORS_ORIGIN_WHITELIST
['stage.edx.org', 'publisher.edx.org']
```

**Comment by awais786** [comment]
With `django-cors-headers==3.2.0`

```
>>> from django.conf import settings
>>> settings.CORS_ORIGIN_WHITELIST
['https://stage.edx.org', 'https://publisher.edx.org']
```

File: course_discovery/settings/production.py [hunk 1]
```diff
@@ -37,6 +38,11 @@
 
     vars().update(config_from_yaml)
 
+    # values are already updated above with default values but in
+    # case of new version they will get override.
+    if version('django_cors_headers') == '3.2.0':
+        vars()['CORS_ORIGIN_WHITELIST'] = vars()['CORS_ORIGIN_WHITELIST_NEW']
```
Comment (waheedahmed) [inline course_discovery/settings/production.py]: As mentioned on the internal vars PR, rename `CORS_ORIGIN_WHITELIST_NEW` to `CORS_ORIGIN_WHITELIST_WITH_SCHEME`.

**Review by waheedahmed — APPROVED** [review]
Looks good 👍

**Review by mraarif — APPROVED** [review]

**Pull request merged**